map_model/src/pathfind/v1.rs

```diff
@@ -514,12 +520,35 @@ impl PathRequest {
         start: Position,
         end: Position,
         constraints: PathConstraints,
-        _: &Map,
+        map: &Map,
     ) -> PathRequest {
+        let alt_start = (|| {
+            let start_lane = map.get_l(start.lane());
+            let road = map.get_r(start_lane.parent);
+            // If start and end road match, don't exit offside
+            // TODO Sometimes this is valid! Just not if we're trying to go behind ourselves
+            if road.id == map.get_l(end.lane()).parent {
+                return None;
+            }
+            let offside_dir = start_lane.dir.opposite();
+            let alt_lane = road.find_closest_lane(
+                start_lane.id,
+                |l| l.dir == offside_dir && constraints.can_use(l, map),
+                map,
+            )?;
+            // TODO Do we need buffer_dist like driving_connection does?
+            let pos = start.equiv_pos(alt_lane, map);
+            let number_lanes_between =
+                ((road.offset(start_lane.id) as f64) - (road.offset(alt_lane) as f64)).abs();
+            // TODO Tune the cost of cutting across lanes
```

Comment:
This works fine in practice gridlock-wise. It would be cool to make this decision right when the vehicle tries to exit the driveway based on current traffic, but that's pushing into dynamic rerouting territory.